macos/builder_setup.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -e
```

Comment:
nit: This is ignored if you run the scripts by doing `bash ./my_script.sh`, while `set -e` works in both cases. Probably not an issue here, but in general I think the later is safer.

Reply:
I think it does matter here, in `datadog-agent-macos-build` we do `bash ./script.sh`. However, I tried a run with `set -e`, and it seems we have some non-critical errors that we'd need to fix (see https://github.com/DataDog/datadog-agent-macos-build/runs/3029648487?check_suite_focus=true) that are likely due to a caching issue.

I'll add a `FIXME` here and in `datadog-agent-macos-build` to explain why we can't set `set -e` right now.